<!--StartFragment-->

<h1 align="center">
HackBio INTERNSHIP 2024
<br/>
<kbd>TRACK: AMR IN CANCER</kbd>
<br/>
</h1>

<p align="center"> HackBio is a global bioinformatics internship that provides hands-on experience in genomics and computational biology, helping participants build practical skills through real-world projects and collaborative learning.</p>

<h2> STAGE 0 </h2>

**TASK** : ESSAY WRITING ON TOPIC "*Drug resistance in cancer.*"

**Skills Developed** : Scientific Writing, Markdown Language.

The task provided insights into the different types of resistance in cancer and the mechanisms underlying drug resistance.

<h2> STAGE 1 </h2>

**TASK** : REVIEW THE PAPER "*Whole-Genome Sequencing Accurately Identifies Resistance to Extended Spectrum Î²-Lactams for Major Gram-Negative Bacterial Pathogens.*"

**Skills developed** : Scientific Writing, Presentation skills.

The task enhanced understanding of the significance of whole-genome sequencing (WGS) for identifying antimicrobial resistance (AMR) in clinical settings and how its evolution could improve the feasibility of genome-wide analysis in clinical microbiology.

<h2> STAGE 2 </h2>

**TASK** : DATA ANALYSIS AND VISUALIZATION OF "*Pipeline of antimicrobial agents in clinical development (WHO - Nov 2021).*"

**Skills developed** : Data Analysis and Visualization, R (Programming language).

The pipeline included information on drug type, clinical development phase, activity against pathogens, innovativeness, route of administration, antibiotic class, and developers of various drug candidates targeting WHO-identified bacterial priority pathogens, including Mycobacterium tuberculosis and Clostridium difficile. This task involved analyzing, visualizing, and presenting the data from the pipeline in the form of infographics.

<h2> STAGE 3 </h2>

**TASK** : **(A)** CODE IN BASH **(B)** DATA ANALYSIS AND VISUALIZATION OF "*Global cholera outbreaks - 1949-2016 (WHO).*"

**Skills developed** : Data Analysis and Visualization, Bash (Scripting language).

The task provided knowledge on handling genomic data using Bash.

<h2> STAGE 4 </h2>

**TASK** : DEVELOP A PIPELINE FOR NGS ANALYSIS.

**Skills developed** : Bash (Scripting language), Google Cloud Shell.

This task involved developing an NGS analysis pipeline using tools such as FASTQC, FASTP, BBMap, BWA, Samtools, and BCFtools for quality control, read alignment, variant calling, and post-alignment processing of genomic data.
